File: Leetcode/1-Two-Sum/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        if (nums.size() < 2) {
            return res;
        }
        map<int, int> hashMap;
        hashMap.clear();
        for (int i = 0; i < nums.size(); i++) {
            int key = target - nums[i];
            if (hashMap.find(key) != hashMap.end()) {
                res.push_back(hashMap[key]);
                res.push_back(i);
                return res;
            }
            hashMap[nums[i]] = i;
        }
        return res;
    }
};

File: Leetcode/1-Two-Sum/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var res = [];
    var map = {};
    for (var i = 0; i < nums.length; i++) {
        var key = target - nums[i];
        if (!map[key]) {
            res.push(map[key], i);
            return res;
        }
        map[nums[i]] = i;
    }
    return res;
};
